staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -151,41 +157,54 @@ func (a *AdmissionOptions) Validate() []error {
 	errs := []error{}
 
 	registeredPlugins := sets.NewString(a.Plugins.Registered()...)
-	for _, name := range a.PluginNames {
+	for _, name := range a.EnablePlugins {
 		if !registeredPlugins.Has(name) {
-			errs = append(errs, fmt.Errorf("admission-control plugin %q is invalid", name))
+			errs = append(errs, fmt.Errorf("enable-admission-plugins plugin %q is unknown", name))
 		}
 	}
 
+	for _, name := range a.DisablePlugins {
+		if !registeredPlugins.Has(name) {
+			errs = append(errs, fmt.Errorf("disable-admission-plugins plugin %q is unknown", name))
+		}
+	}
+
+	enablePluginSet := sets.NewString(a.EnablePlugins...)
+	disablePluginSet := sets.NewString(a.DisablePlugins...)
+	if len(enablePluginSet.Intersection(disablePluginSet).List()) > 0 {
+		errs = append(errs, fmt.Errorf("enable-admission-plugins %q and disable-admission-plugins %q "+
+			"must not contain same plugins", a.EnablePlugins, a.DisablePlugins))
+	}
+
+	// Verify RecommendedPluginOrder.
+	if sets.NewString(a.RecommendedPluginOrder...).Len() != len(a.Plugins.Registered()) {
+		// Developer error, this should never run in.
+		errs = append(errs, fmt.Errorf("the number of registered admission plugins not equal to that of in the ordered list"))
+	}
+
 	return errs
 }
 
-// enabledPluginNames makes use of RecommendedPluginOrder and DefaultOffPlugins fields
+// enabledPluginNames makes use of RecommendedPluginOrder, DefaultEnabledPlugins,
+// EnablePlugins, DisablePlugins, PluginNames fields
 // to prepare a list of plugin names that are enabled.
-//
-// TODO(p0lyn0mial): In the end we will introduce two new flags:
-// --disable-admission-plugin this would be a list of admission plugins that a cluster-admin wants to explicitly disable.
-// --enable-admission-plugin  this would be a list of admission plugins that a cluster-admin wants to explicitly enable.
-// both flags are going to be handled by this method
 func (a *AdmissionOptions) enabledPluginNames() []string {
-	//TODO(p0lyn0mial): first subtract plugins that a user wants to explicitly enable from allOffPlugins (DefaultOffPlugins)
-	//TODO(p0lyn0miial): then add/append plugins that a user wants to explicitly disable to allOffPlugins
-	//TODO(p0lyn0mial): so that --off=three --on=one,three default-off=one,two results in  "one" being enabled.
-	allOffPlugins := a.DefaultOffPlugins
```

Comment:
This code here is correct.  All you need to do is 
```
allOffPlugins := append(a.DefaultOffPlugins.List(), a.DisabledPlugins)
```

right?  You've already confirmed all the names are valid and you've confirmed the sets don't overlap, so the remainder of this logic can remain the same.

Reply:
I see.